KC7-Foundation/kc7_scoreboard
# Reformatted leaderboard and view

### Description

This PR introduces a reformatted version of the leaderboard page and its view. The page has been simplified by removing large headings that stick out. The `leaderboard()` view has been rewritten to enhance simplicity and efficiency.

### Changes made

1. `views.py`
   - Removed the caching decorator `@cache.memoize` and the nested function `get_score()`. Instead, directly called the `get_score_matrix_simple()` function to retrieve the score matrix.
   - Simplified the sorting process by using the `nlargest()` function to get the IDs of the top 10 users based on scores.
   - Retrieved the user objects using the retrieved user IDs in a single database query.
   - Sorted the users list in descending order based on their scores.

2. `leaderboard.html/custom.css`
   - Simplified the page and removed unneeded classes, primarily using Bootstrap classes.
   - Simplified `leaderboard_link` CSS and made it a class instead of an ID to avoid duplicate ID errors.

### How to test

1. Navigate to the leaderboard page.
2. Ensure that the table looks okay.
3. Verify that the custom CSS remains intact when hovering over the usernames.

### Screenshots

| Before | After |
| ------ | ----- |
| ![Before](https://github.com/KC7-Foundation/kc7_scoreboard/assets/97637517/bb7bfaa3-c8d0-4e16-af0a-706941c26718) | ![image](https://github.com/KC7-Foundation/kc7_scoreboard/assets/97637517/fd54aa26-0e68-447c-95c1-461f15b3b384) |

Future Improvements: I would like to center the usernames column, but I couldn't find a solution that aligns the robot icons correctly.
